WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayInformationViewController.swift

```diff
@@ -0,0 +1,189 @@
+        return BuiltInCardReaderConnectionController(
+            forSiteID: viewModel.siteID,
+            alertsPresenter: alertsPresenter,
+            alertsProvider: BuiltInReaderConnectionAlertsProvider(),
+            configuration: viewModel.configuration,
+            analyticsTracker: viewModel.connectionAnalyticsTracker)
+    }()
+
+    init?(viewModel: PaymentSettingsFlowPresentedViewModel) {
+        guard let viewModel = viewModel as? SetUpTapToPayInformationViewModel else {
+            return nil
+        }
+        self.viewModel = viewModel
+
+        super.init(rootView: SetUpTapToPayInformationView())
+        configureView()
+    }
+
+    private func configureView() {
+        rootView.setUpButtonAction = { [weak self] in
+            self?.searchAndConnect()
+        }
+        rootView.showURL = { url in
+            WebviewHelper.launch(url, with: self)
+        }
+        rootView.learnMoreUrl = viewModel.learnMoreURL
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("Not implemented")
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        viewModel.didUpdate = nil
+        super.viewWillDisappear(animated)
+    }
+}
+
+// MARK: - Convenience Methods
+//
+private extension SetUpTapToPayInformationViewController {
+    func searchAndConnect() {
+        connectionController.searchAndConnect { _ in
+            /// No need for logic here. Once connected, the connected reader will publish
+            /// through the `cardReaderAvailableSubscription`
+            self.alertsPresenter.dismiss()
+        }
+    }
+}
+
+struct SetUpTapToPayInformationView: View {
+    var setUpButtonAction: (() -> Void)? = nil
+    var showURL: ((URL) -> Void)? = nil
+    var learnMoreUrl: URL? = nil
+
+    @Environment(\.verticalSizeClass) var verticalSizeClass
+    @Environment(\.sizeCategory) private var sizeCategory
+
+    var isCompact: Bool {
+        get {
+            verticalSizeClass == .compact
+        }
+    }
+
+    var isSizeCategoryLargerThanExtraLarge: Bool {
+        sizeCategory >= .accessibilityMedium
+    }
+
+    var body: some View {
+        VStack {
+            Spacer()
+
+            Text(Localization.setUpTapToPayOnIPhoneTitle)
+                .font(.title.weight(.semibold))
+                .padding(32)
+                .multilineTextAlignment(.center)
+            Image(uiImage: .setUpBuiltInReader)
+                .resizable()
+                .scaledToFit()
+                .frame(height: isCompact ? 80 : 206)
+                .padding(.bottom, isCompact ? 16 : 32)
+
+            PaymentSettingsFlowHint(title: Localization.hintOneTitle, text: Localization.hintOne)
+            PaymentSettingsFlowHint(title: Localization.hintTwoTitle, text: Localization.hintTwo)
+            PaymentSettingsFlowHint(title: Localization.hintThreeTitle, text: Localization.hintThree)
+
+            Spacer()
+
+            Button(Localization.setUpButton, action: setUpButtonAction!)
```

Comment:
This closure will be nil by default unless it's set, is that right? We could do a nil-check here to avoid the force-unwrapping:

```suggestion
            Button(Localization.setUpButton, action: {
                if let setUpButtonAction = setUpButtonAction {
                    setUpButtonAction()
                }
            })
```

Reply:
👍  [Already done, but in the next PR](https://github.com/woocommerce/woocommerce-ios/pull/9036/files)